mithril-common/src/test_utils/temp_dir.rs

```diff
@@ -0,0 +1,232 @@
+use sha2::{Digest, Sha256};
+use std::path::{Path, PathBuf};
+
+/// A builder of temp directory for tests purpose.
+#[derive(Clone)]
+pub struct TempDir {
+    module_name: String,
+    name: String,
+    enable_short_path: bool,
+    short_path_max_len: usize,
+}
+
+const TEMP_DIR_ROOT_NAME: &str = "mithril_test";
+// 90 to have some room for the folder content (in case of restrained length like socket path)
+const DEFAULT_SHORT_PATH_MAX_LEN: usize = 90;
+
+impl TempDir {
+    /// `TempDir` builder factory
+    pub fn new<T: Into<String>>(module: T, name: T) -> Self {
+        Self {
+            module_name: module.into(),
+            name: name.into(),
+            enable_short_path: false,
+            short_path_max_len: DEFAULT_SHORT_PATH_MAX_LEN,
+        }
+    }
+
+    /// Change path generation in order to guarantee a path that have at maximum 90 characters.
+    ///
+    /// Typically used for cases when the generated folder will include a socket.
+    pub fn generate_shorter_path(mut self) -> Self {
+        self.enable_short_path = true;
+        self
+    }
+
+    /// Set the max len that a short path can have
+    pub fn set_short_path_max_len(mut self, max_len: usize) -> Self {
+        self.short_path_max_len = max_len;
+        self
+    }
+
+    /// Generate the path of the temp directory (no IO operation will be executed)
+    pub fn build_path(&self) -> PathBuf {
+        const SHA_LENGTH: usize = 10;
+        let base_dir = std::env::temp_dir().join(TEMP_DIR_ROOT_NAME);
+
+        // Short path only:
+        // Combined max len should be lower than `self.short_path_max_len` to have some rooms for
+        // the folder content.
+        // MacOS temp folder are not in the `/tmp` folder but in a dynamic path adding 45 chars.
+        // ie: /var/folders/_k/7j0m5c_n4g94vgx9gxknp4tm0000gn/T/
+        if self.enable_short_path {
+            // In order to discriminate two tests with the same name but within different modules
+            // we append the short sha of the module+name to the path.
+            let mut hasher = Sha256::new();
+            hasher.update(&self.module_name);
+            hasher.update(&self.name);
+            let sha = hex::encode(hasher.finalize());
+            let short_sha = &sha[0..SHA_LENGTH];
+
+            // `-2` since when joining a path this adds a `/` and we also add a `_` to join the sha
+            let max_path_len =
+                self.short_path_max_len - SHA_LENGTH - 2 - base_dir.to_string_lossy().len();
+
+            let max_len = self.name.len().min(max_path_len);
+
+            base_dir.join([&self.name[0..max_len], "_", short_sha].concat())
+        } else {
+            base_dir.join(&self.module_name).join(&self.name)
+        }
+    }
+
+    /// Create a directory based of builder configuration in the system temp folder.
```

Comment:
```suggestion
    /// Create a directory based on the builder configuration in the system temp folder.
```

Also, the expected behavior if the directory already exists could be documented here.